Neuro/Minigames/Completion/Solvers/PrimeShieldsSolver.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections;
+using Neuro.Cursor;
+
+namespace Neuro.Minigames.Completion.Solvers;
+
+[MinigameSolver(typeof(ShieldMinigame))]
+public sealed class PrimeShieldsSolver : MinigameSolver<ShieldMinigame>
+{
+    public override IEnumerator CompleteMinigame(ShieldMinigame minigame, NormalPlayerTask task)
+    {
+        for (int i = 0; i < minigame.Shields.Count; i++)
+        {
+            if (minigame.Shields[i].color == minigame.OffColor)
```

Comment:
Instead of checking for the color, use `minigame.shields`, it is a bitmask.

Example of usage from `ShieldMinigame.ToggleShield`:
```cs
public void ToggleShield(int i)
{
	if (!this.MyNormTask.IsComplete)
	{
		byte b = (byte)(1 << i);
		this.shields ^= b;
		this.MyNormTask.Data[0] = this.shields;
		if ((this.shields & b) != 0)
		{
			if (Constants.ShouldPlaySfx())
			{
				SoundManager.Instance.PlaySound(this.ShieldOnSound, false, 1f, null);
			}
		}
		else if (Constants.ShouldPlaySfx())
		{
			SoundManager.Instance.PlaySound(this.ShieldOffSound, false, 1f, null);
		}
		if (this.shields == 127)
		{
			this.MyNormTask.NextStep();
			base.StartCoroutine(base.CoStartClose(0.75f));
			if (ShipStatus.Instance.ShieldsImages.Length != 0 && !ShipStatus.Instance.ShieldsImages[0].IsPlaying(null))
			{
				PlayerControl.LocalPlayer.RpcPlayAnimation(1);
			}
		}
	}
}
```